addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthUtils.java

```diff
@@ -1069,7 +1072,7 @@ public static void recordRequestSessionToken(Context context, String key, String
 
     public static void recordRequestSessionToken(int contextId, String key, String value) {
         requestTokenMap
-                .computeIfAbsent(contextId, c -> new HashMap<>())
+                .computeIfAbsent(contextId, c -> Collections.synchronizedMap(new HashMap<>()))
```

Comment:
Better extract a method.

Reply:
I was referring to both lines not just the the internal map creation.

Reply:
Adjusted